internal/mode/static/telemetry/collector_test.go

```diff
@@ -363,4 +404,106 @@ var _ = Describe("Collector", Ordered, func() {
 			})
 		})
 	})
+
+	Describe("NGF replica count collector", func() {
+		When("collecting NGF replica count", func() {
+			When("it encounters an error while collecting data", func() {
```

Comment:
Looks like you need a test for the success case.

Reply:
The tests are set up in a kind of not the most clear way where the first Describe "Normal case" runs through and collects all the telemetry data (including the deployment replica count) while the following (what you commented on) are for "edge cases". 

So the first Describe case is the success case where the replica = 1, and I couldn't think of a need to have another case which tests for a different replica amount so all I thought we needed here were the edge cases of erroring.

Reply:
Yep you're right. My brain focused in too much on those edge cases and forgot about the earlier one.